### PR TITLE
feat(avatar): group variant

### PR DESCRIPTION
**Pull request by francisrupert**
## Description

* added `group` variant
* added `no-gradient` variant
* refactored avatar html structure

image

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/components/avatar.md [hunk 1]
```diff
@@ -10,19 +10,29 @@ figma_url: https://www.figma.com/file/2adf7JhZOncRyjYiy2joil/DT-Core%3A-Componen
...
             </div>
         </div>
         <div class="d-avatar d-avatar--md">
-            <svg aria-hidden="true" focusable="false" data-name="Video" class="d-icon d-icon--video d-icon--size-300" viewBox="0 0 12 12" data-v-68a9d516=""><path fill="currentColor" fill-rule="evenodd" d="M2 3.5a.5.5 0 0 0-.5.5v4a.5.5 0 0 0 .5.5h5a.5.5 0 0 0 .5-.5V4a.5.5 0 0 0-.5-.5H2Zm6.5 1.566V4A1.5 1.5 0 0 0 7 2.5H2A1.5 1.5 0 0 0 .5 4v4A1.5 1.5 0 0 0 2 9.5h5A1.5 1.5 0 0 0 8.5 8V6.934l2.223 1.482A.5.5 0 0 0 11.5 8V4a.5.5 0 0 0-.777-.416L8.5 5.066ZM8.901 6 10.5 7.066V4.934L8.901 6Z" clip-rule="evenodd"></path></svg>
+            <div class="d-avatar__canvas">
+                <span class="d-avatar__icon">
+                    <span class="d-avatar__icon">
+                        <dt-icon name="video" size="300" />
+                    </span>
+                </span>
```
Comment (josedialpad) [inline docs/components/avatar.md]: Remove extra `d-avatar__icon` wrapper

```suggestion
                <span class="d-avatar__icon">
                      <dt-icon name="video" size="300" />
                </span>
```

File: docs/components/avatar.md [hunk 2]
```diff
@@ -221,6 +377,40 @@ Unique gradient combinations are generated by the Vue component. It randomizes s
 </div>
 ```
 
+### Group
+
+<code-well-header>
```
Comment (josedialpad) [inline docs/components/avatar.md:380]: Can be useful to exemplify the usage of the group count with the different sizes as it's in the presence example?

**Review by josedialpad — COMMENTED** [review]
Great work, approved w/ 2 suggestions. Will give the formal approval when the vue implementation is updated as this avatar [here](https://dialpad.atlassian.net/browse/DT-891) as this update contains some breaking changes.

File: docs/components/avatar.md (hunk 1 above)
Comment (francisrupert) [reply docs/components/avatar.md]: Good catch.

File: docs/components/avatar.md (hunk 2 above)
Comment (francisrupert) [reply docs/components/avatar.md:380]: One size only as acknowledged this morning.

**Comment by francisrupert** [comment]
@josedialpad I see that you've got [DT-891](https://dialpad.atlassian.net/browse/DT-891) in progress. I'll defer to you and @braddialpad for timing the approval and merge of this to staging as it relates to timing the release of both `dialtone` and `dialtone-vue`.

[DT-891]: https://dialpad.atlassian.net/browse/DT-891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by josedialpad** [comment]
> @josedialpad I see that you've got [DT-891](https://dialpad.atlassian.net/browse/DT-891) in progress. I'll defer to you and @braddialpad for timing the approval and merge of this to staging as it relates to timing the release of both `dialtone` and `dialtone-vue`.

Yeah, created a [draft PR](https://github.com/dialpad/dialtone-vue/pull/745) for the vue changes, once the vue2 and vue3 PRs are approved we can move forward with merging and releasing.

[DT-891]: https://dialpad.atlassian.net/browse/DT-891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by josedialpad** [comment]
@francisrupert I have a case when wrapping the avatar with presence with a non-flex element (for example a raw div). The positioning of the presence element is going out of the avatar.
Repro: I've deployed your branch to [v7.16.0-alpha.1](https://www.npmjs.com/package/@dialpad/dialtone/v/7.16.0-alpha.1) and updated the vue PR with this version, so setting the [presence in the default avatar](https://vue.dialpad.design/deploy-previews/pr-745/?path=/story/components-avatar--default&args=presence:active) makes the presence go to the right

**Comment by francisrupert** [comment]
@josedialpad [21fc2ed](https://github.com/dialpad/dialtone/pull/813/commits/21fc2ed6801e94ec3213da92914d259c4d72b1dc) resolves.

**Comment by braddialpad** [comment]
@francisrupert Do we need to publish a blog post on breaking changes of this avatar?

It seems that new wrapper class `d-avatar__canvas` would have to be added to any HTML/CSS usages of avatar. I think the vue component changes will be non-breaking. I think this is the only one since pretty much everything else is net new.

**Comment by francisrupert** [comment]
@braddialpad Yeah, that would be wise.

**Comment by braddialpad** [comment]
Blog post PR up here https://github.com/dialpad/dialtone/pull/816

**Review by braddialpad — APPROVED** [review]
Jose's PR is now approved this is good to merge.

**Pull request merged**